nixos/modules/services/networking/prosody.nix

```diff
@@ -390,6 +390,54 @@ let
     };
   };
 
+  pubsubOpts = { ... }: {
+    options = {
+      domain = mkOption {
+        type = types.nullOr types.str;
+        description = lib.mdDoc "Domain name for the pubsub service";
+      };
+
+      # https://prosody.im/doc/modules/mod_pubsub
+      admins = mkOption {
+        type = types.listOf types.str;
+        default = [ ];
+        example = [ "admin1@example.com" "admin2@example.com" ];
+        description = lib.mdDoc "List of administrators of the current pubsub host";
+      };
+      defaultAdminAffiliation = mkOption {
+        type = types.str;
+        default = "owner";
+        description = lib.mdDoc "The default affiliation of admins to nodes";
+      };
+      autocreateOnSubscribe = mkOption {
+        type = types.bool;
+        default = false;
+        description = lib.mdDoc "Whether to automatically create non-existent nodes when someone subscribes";
+      };
+      autocreateOnPublish = mkOption {
+        type = types.bool;
+        default = false;
+        description = lib.mdDoc "Whether to automatically create non-existent nodes when someone publishes";
+      };
+      maxItems = mkOption {
```

Comment:
This is called `pubsub_max_items` in the config, but that seems so redundant as `cfg.pubsub.pubsubMaxItems`.